docs/move/book/loops.md

```diff
@@ -114,6 +114,45 @@ fun pick(
 }
 ```
 
+## The `for` expression
+
+The `for` expression iterates over a range defined using integer-typed `lower_bound` (inclusive) and `upper_bound` (non-inclusive) expressions, executing its loop body for each element of the range. `for` is designed for scenarios where the number of iterations of a loop is determined by a specific range.
+
+Here is an example of a `for` loop that computes the sum of the elements in a range from `0` to `n-1`:
+
+```move
+fun sum(n: u64): u64 {
+    let sum = 0;
+    for (i in 0..n) {
+        sum = sum + i;
+    };
+
+    sum
+}
+```
+
```

Comment:
Suggest adding some details:

The loop index variable (`i` in this example) currently must be a numeric type (inferred from the bounds), and the bounds `0` and `n` here can be replaced by arbitrary numeric expressions.  Each is only evaluated once, at the start of the loop.  The index variable `i` is assigned the starting bound `0` and incremented after each loop iteration; when the index `i` reaches or exceeds the upper bound than the loop exits.

Reply:
Added the description.  I am calling it an iterator as it is not necessarily a loop index (for example when lower bound is non-zero).